site/docs/spec.md

```diff
@@ -176,6 +178,15 @@ Columns in Iceberg data files are selected by field id. The table schema's colum
 For example, a file may be written with schema `1: a int, 2: b string, 3: c double` and read using projection schema `3: measurement, 2: name, 4: a`. This must select file columns `c` (renamed to `measurement`), `b` (now called `name`), and a column of `null` values called `a`; in that order.
 
 
+#### Identifier Field IDs
+
+A schema can optionally track the set of primitive fields that identify rows in a table, using the property `identifier-field-ids` (see JSON encoding in Appendix C).
+
+Two rows are the "same"---that is, the rows represent the same entity---if the identifier fields are equal. However, uniqueness of rows by this identifier is not guaranteed or required by Iceberg because it is needlessly inefficient in some cases. Reasonable steps to enforce uniqueness are the responsibility of processing engines.
```

Comment:
I feel like we should change the wording of the second sentence.  I don't think "needlessly inefficient" is a great way to represent what we're trying to convey.  Maybe just:  However, uniqueness of rows by this identifier is not guaranteed or required by Iceberg and is the responsibility of processing engines or data producers to enforce.

Reply:
Updated to use your suggested wording.